DBAgenda.php

```diff
@@ -0,0 +1,311 @@
+        $query->execute(array(
+            'vCalendarFilename' =>  $event['vCalendarFilename'],
+            'ETag' => $event['ETag'],
+            'datetime_begin' => $datetime_begin->format('Y-m-d H:i:s'),
+            'number_volunteers_required' => $number_volunteers_required,
+            'vCalendarRaw' => $event['vCalendarRaw']
+        ));
+        
+        if(isset($vCalendar->VEVENT->ATTENDEE)) {
+            foreach($vCalendar->VEVENT->ATTENDEE as $attendee) {
+                $mail = str_replace("mailto:", "", (string)$attendee);
+
+                $query = $this->pdo->prepare("SELECT * FROM attendees WHERE email=:email;");
+                $query->execute(array(
+                    'email' => $mail
+                ));
+
+                if(is_array($ret = $query->fetch())) {
+                    $this->log->debug("attendee: $mail already exists.");
+                } else {
+                    $user = $this->api->users_lookupByEmail($mail);
+                    if(!is_null($user)) {
+                        $userid = $user->id;
+                    } else {
+                        $userid = null;
+                    }
+
+                    $query = $this->pdo->prepare("REPLACE INTO attendees (email, userid) VALUES (:email, :userid)");
+                    $query->execute(array(
+                        'email' =>  $mail,
+                        'userid' =>  $userid
+                    ));
+                }
+                
+                $query = $this->pdo->prepare("INSERT INTO events_attendees (vCalendarFilename, email) VALUES (:vCalendarFilename, :email)");
+                $query->execute(array(
+                    'vCalendarFilename' =>  $event['vCalendarFilename'],
+                    'email' =>  $mail
+                ));
+            }
+        }
+        
+        if(isset($vCalendar->VEVENT->CATEGORIES)) {
+            foreach($vCalendar->VEVENT->CATEGORIES as $category) {
+                $category = (string)$category;
+                
+                if(is_number_of_attendee_category($category)) {
+                    continue;
+                }
+                
+                $query = $this->pdo->prepare("SELECT * FROM categories WHERE name=:name;");
+                $query->execute(array(
+                    'name' => $category
+                ));
+                
+                $id = null;
+                if(is_array($ret = $query->fetch())) {
+                    $id = $ret['id'];
+                    $this->log->debug("category: $category already exists.");
+                } else {                    
+                    $query = $this->pdo->prepare("INSERT INTO categories (name) VALUES (:name);");
+                    $query->execute(array(
+                        'name' => $category
+                    ));
+                    
+                    $id = $this->getLastInsertedRowId();
```

Comment:
I think we may have race condition here, in the case where 2 INSERTs are done at almost the same time, then one execution may end up with the $id of the other.
I'm not sure what would be the best way to tackle it. Perhaps using some sql transation?